server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java

```diff
@@ -586,4 +572,33 @@ private Runnable scheduleTimeout(TimeValue timeout, ThreadPool threadPool, Execu
     private Object compareAndExchangeState(Object expectedValue, Object newValue) {
         return VH_STATE_FIELD.compareAndExchange(this, expectedValue, newValue);
     }
+
+    @SuppressWarnings("rawtypes")
+    private static final SubscribableListener NULL_SUCCESS = newSucceeded(null);
+
+    /**
+     * Same as {@link #newSucceeded(Object)} but always returns the same instance with result value {@code null}.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> SubscribableListener<T> nullSuccess() {
+        return NULL_SUCCESS;
+    }
+
+    /**
+     * Create a {@link SubscribableListener} which has already succeeded with the given result.
+     */
+    public static <T> SubscribableListener<T> newSucceeded(T result) {
+        var res = new SubscribableListener<T>();
```

Comment:
This allocates a `new Object()` for the initial state, can we find a way not to do that?

Reply:
Ah 🤦 gotta fix that too, otherwise this is kinda pointless. Sec on it

Reply:
Alright, not sure you'll like this one :) But if I get rid of `EMPTY` this all falls into place just fine doesn't it?